content/en/docs/tasks/run-application/configure-pdb.md

```diff
@@ -277,7 +277,8 @@ Pods in `Pending`, `Succeeded` or `Failed` phase are always considered for evict
 ## Arbitrary Controllers and Selectors
 
 You can skip this section if you only use PDBs with the built-in
-application controllers (Deployment, ReplicationController, ReplicaSet, and StatefulSet),
+application controllers (Deployment, ReplicationController, ReplicaSet, and StatefulSet)
+or controllers that implement a [scale subresource](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource),
```

Comment:
```suggestion
You can skip this section if you only use PDBs with the built-in
workload management object kinds (Deployment, ReplicaSet, StatefulSet and ReplicationController),
or with {{< glossary_tooltip term_id="CustomResourceDefinition" text="custom resources" >}} that
implement a `scale` [subresource](/docs/concepts/extend-kubernetes/api-extension/custom-resources/#advanced-features-and-flexibility).
```

We tend not to call Deployment a controller. One reason is that if you call something ReplicationController, and then you have a control loop that takes the `.spec` of a ReplicationController as its desired state, you then kind of have to call the code the “ReplicationController controller”.

I'd prefer to revise the wording whilst we're touching this section. Also, I propose adding a glossary tooltip to explain CRD.

Reply:
I guess it does not make that much sense to reference it as a controller from user's perspective here.

Workload management object kinds sounds too technical/specific. Can we call them workload resources? (we use this term in the concepts as well https://kubernetes.io/docs/concepts/workloads/controllers/)

I made the changes to the rest of this section. But we mention controller in the rest of the document. So should we rewrite it all, or do a followup with a focus on that?

+1 for the glossary.

Regarding the scale subresource link - the proposed one gives a high level definition which links to the more detailed one. Not sure which one is better here.

Nevertheless we probably want to be consistent with the link in the intro, right? Should I swap it as well?

Reply:
I am going back and forth regarding the controller vs workload resource.

The pod should be owned by a resource which is a controller ([ownerReferences.controller](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#System)), so from k8s perspective the ReplicaSet is the controller and I am not sure if we want to diverge from that.